core/sql/cli/SessionDefaults.cpp

```diff
@@ -108,6 +108,7 @@ static const SessionDefaults::SessionDefaultMap sessionDefaultMap[] =
   SDEntry(SessionDefaults::SCHEMA,                   SCHEMA,                     SessionDefaults::SDT_ASCII,          TRUE,    TRUE,  FALSE, FALSE),
   SDEntry(SessionDefaults::STATISTICS_VIEW_TYPE,     STATISTICS_VIEW_TYPE,       SessionDefaults::SDT_ASCII,          FALSE,   FALSE, TRUE,  TRUE),
   SDEntry(SessionDefaults::SUSPEND_LOGGING,          SUSPEND_LOGGING,            SessionDefaults::SDT_BOOLEAN,        FALSE,   FALSE, TRUE,  FALSE),
+  SDEntry(SessionDefaults::USE_LIBHDFS,              USE_LIBHDFS,                SessionDefaults::SDT_BOOLEAN,        TRUE,    TRUE,  FALSE, FALSE),
```

Comment:
There is already a CQD USE_LIBHDFS, do we also need a session attribute? Also there are some CQDs that can't be set in the ```"_MD_".DEFAULTS``` table. I haven't checked this but hope that USE_LIBHDFS isn't one of them.

Reply:
There is a need to access this CQD value in CLI. Hence, this CQD is converted into SSD. Setting this CQD converts into SSD automatically. See executor/ex_control.cpp. However, the settings of this SSD is such that it is not settable via SSD command. It can be set only via CQD.
I would want USE_LIBHDFS to be settable in the defaults table. Unfortunately, the CQD set in the defaults table is not effective in CLI layer because it is not translated into SSD.  I would be fixing it in a separate PR if it is really needed at a future date.

Reply:
Thanks for the explanation. Don't you think that it will be a requirement to make this work with the defaults table? Maybe an environment variable would be a better choice if the defaults table doesn't work?

Reply:
The CQD could have been passed in just like we pass in LOB_MAX_SIZE and LOB_MAX_MEM_CHUNK size but that makes the code a bit messy and Selva will need to change way too many interfaces. The CQD as SSD is a good choice for this particular case since it's a "setting".  It's better not to have it as an envvar since switching back and forth  will make it more difficult.

Reply:
I found that it wasn't possible to do like CQD LOB_MAX_SIZE because this setting is needed in ExLOBGlobals constructor.  The best way is to transform it into SSD, if the settings is needed in CLI layer.  Then I realized that CQD is effective but SSD isn't effective if the CQD is set in defaults table.  Currently if you set this CQD to 'ON' in defaults tables, the portions of the code that created ExLobGlobals from CLI layer still uses newer implementation. 

USE_LIBHDFS is just fall back option till such time the new implementation is tested.  Then, I would think the code and the CQD can be retired. I haven't it figured out how much impact is not to use USE_LIBHDFS settings from the CLI layer when we need to fallback. I have run core/TEST130 with CQD USE_LIBHDFS 'ON' in defaults table. 

Because it is a generic issue of not transforming to SSD from CQD when it is defaults table, I thought that I will handle it via another PR. In addition, this would require changes to an area that I am not yet comfortable with.

Reply:
Yes, fixing all those CQDs to allow entries in the system defaults table would be nice. But, that will run into some design issues: CQDs affect compilation, session attributes affect the runtime, so there is a mismatch in the scope. I would prefer a solution that respects those designs. Would having a flag word in the statement globals help?

You say that the USE_LIBHDFS CQD is just a fallback, but that is exactly what won't work if we don't support the system defaults table. Let's say a user would have to re-enable libhdfs because of some problem. If they can't do that in the system defaults table, they would have to change all their affected programs, which is not going to make them very happy. If it's just for fallback in case of problems, I think an environment variable will be better for users. I know that the CQD is of course better for our own testing.

Reply:
+1
While I do think the CQD is a problem, given that this is a temporary situation we should be able to live with the current state of things. If all fails we may need to produce a code patch.

Reply:
I am confused by this now. Are we saying if we set the CQD USE_LIBHDFS to ON, the code today will not revert to the old behavior ?   Given several POCs are using this feature and usually under a time crunch, it will not be practical  provide a patch. I think we need a mechanism to revert to the old interface for 2 reasons. We also need to run performance tests with very large LOBs. China team is waiting for this and need a way to turn this feature off and on for sure. Maybe envvar is the way to go if this CQD/SSD is only partial.

Reply:
@sandhyasun As a contributor involved with LOB interface, can you please let me know what features of LOB are implemented via CLI calls? Within those CLI calls, which of those feature will inovke LOB functions without compilation of SQL statement?

Reply:
Do you means the different LOB functions/syntax ?  LOB itself is considered one single  feature. 
Different LOB operations are implemented as functions eg filetolob(), buffertolob(), stringtolob()  which get compiled as part of a SQL DML  statement. These get evaluated at runtime .  
And some LOB operations are implemented as Executor utility operations eg :  UPDATE LOB and EXTRACT LOB.

Reply:
The update and extract LOB commands were not adhering to USE_LIBHDFS settings. This has been fixed in commit ee44ad9. However, all the DDL lob functions executed via the CLI call SQL_EXEC_LOBddlInterface will not adhere to USE_LIBHDFS settings if the CQD is set in defaults table.  Instead the CQD should be set via DCS profile till we fix the issue of transforming CQD into SSD for the CQDs in defaults table.  Because it is is DDLinterface, it should be ok even if it couldn't switch to older implementation.

Reply:
It's fine if the DDL interfaces use the new implementation since we just create or drop. Since you took care of the 2 utility operations UPDATE and EXTRACT, looks like we are covered all the main functionality with this toggle now. Looks good +1 .  Can merge as soon as the test passes.